### PR TITLE
:art: consider upcoming and passed meetings (remove necessity for mdx)

### PR DESCRIPTION
**Pull request by romedius**
ok - got rid of the mdx files.

split up the allMetups into a filtered newMeetups array and used passedMeetups for the archive

### CODE REVIEW

File: src/pages/meetup/alps/promote.astro [hunk 1]
```diff
@@ -8,9 +8,10 @@ let description = 'Tech Meetup in Innsbruck, brings together tech enthusiasts, e
 const canonicalURL = new URL(Astro.url.pathname, Astro.site);
 
 // Current Meetup
-let allMeetups = await Astro.glob('./meetups/*.md');
-allMeetups.sort((a, b) => new Date(b.frontmatter.date).valueOf() - new Date(a.frontmatter.date).valueOf());
-const { frontmatter: nextMeetup } = allMeetups[0];
+let allMeetups = await Astro.glob('./meetups/*.md')
+let upcomingMeetups = allMeetups.filter((meetup) => new Date(meetup.frontmatter.date) > new Date());
```
Comment (timhannemann) [inline src/pages/meetup/alps/promote.astro:12]: same as above

**Review by timhannemann — CHANGES_REQUESTED** [review]
perfect, but I think it is important that we show the meetup longer that till the startdate! See my suggestion!

File: src/components/MeetupListing.astro [hunk 2]
```diff
@@ -13,6 +13,12 @@ const { numberOfItems } = Astro.props;
 
 // Build the list of Meetups
 let allMeetups = await Astro.glob('../pages/meetup/alps/meetups/*.md');
+let upcomingMeetups = allMeetups.filter((meetup) => new Date(meetup.frontmatter.date) > new Date());
```
[the remark this answers is not in the document]
Comment (woolfg) [reply src/components/MeetupListing.astro:16]: or maybe even +48 hours?

[the remark this answers is not in the document]
Comment (woolfg) [reply src/components/MeetupListing.astro:16]: I would suggest to move the whole logic of fetching meetups to a separate file providing a fetch function that takes an argument to specify what meetups are needed.

File: src/pages/meetup/alps/index.astro
Comment (woolfg) [inline src/pages/meetup/alps/index.astro:1]: in this file we fetch all information to use on the index page and then use <MeetupListing> which fetches all information again. maybe we can just use an argument on MeetupListing which takes the meetups? the outer scope is then responsible to the fetch the needed meetups.

File: src/components/MeetupListing.astro (hunk 2 above)
[the remark this answers is not in the document]
Comment (timhannemann) [reply src/components/MeetupListing.astro:16]: why 48h?
central fetching and params would be good, yes!

[the remark this answers is not in the document]
Comment (woolfg) [reply src/components/MeetupListing.astro:16]: - if people want to lookup something on the next day, it is still there and not in the archive yet.
- in general it is not exact, as it is just updated when an commit triggers the build of the website, although that happens quite often as we have a lot of triggers and auto-updates.

[the remark this answers is not in the document]
Comment (timhannemann) [reply src/components/MeetupListing.astro:16]: isn't there some nightly workflow running anyway? but yes, some time-delay would be nice, 24 or 48 hours - I don't care...

[the remark this answers is not in the document]
Comment (woolfg) [reply src/components/MeetupListing.astro:16]: there is just a weekly on tuesday, but hopefully we always add the slides on the next day anyways ;)

**Comment by romedius** [comment]
hi!

@woolfg , sag ma einfach auf was ich das einstellen soll - ich modularisiers mal raus und machs irgendwie konfigurierbar.

IMHO: Wenns ne gute Verbesserung ist, im Zweifel mal mergen und dann darauf aufbauen - sowohl modularisierung als auch die Zeitverschiebung :)

**Pull request merged**